lib/dangermattic/plugins/manifest_pr_checker.rb

```diff
@@ -99,6 +99,23 @@ def check_swift_package_resolved_updated_strict(manifest_path:, manifest_lock_pa
       )
     end
 
+    # Check for duplicate Package.resolved files across all provided SwiftPM package directories
+    #
+    # @param report_type [Symbol] (optional) The type of report for the message. Types: :error, :warning (default), :message.
+    # @param spm_root_paths [Array<String>] Array of paths to SwiftPM package root directories (e.g. ['Modules'])
+    #
+    # @return [void]
+    def check_duplicate_package_resolved(spm_root_paths:, report_type: :warning)
+      resolved_files = spm_root_paths.flat_map { |root_path| Dir.glob("#{root_path}/**/Package.resolved") }
+      return if resolved_files.length <= 1
```

Comment:
If I'm understanding the way some repos use SPM, I think it might still be totally valid to have multiple `Package.resolved` files in the repo, when you have multiple (local) SwiftPM packages in a monorepo

For example, a repo could use:
 - An `xcworkspace` at the root of the repo for the main app project, which might have some (local & 3rd party) packages defined in the "Package Dependencies" tab of its Xcode project
 - But also in order to organize the code of the app in submodules, might also have a `Modules/` or `Components/` subfolder, with potentially many local packages in there (either one per sub-component, or a single package defining all the components in a single `Package.swift`).

In such cases I'd expect as many `Package.resolved` as there are package definitions (either in a `Package.swift` or in dependencies defined direclty in a `xcodeproj`'s "Package Dependencies" tab) in total in the repo…

For example, Tumblr/iOS has:
 - [`Orangina.xcworkspace/xcshareddata/swiftpm/Package.resolved`](https://github.tumblr.net/Tumblr/ios/tree/develop/Orangina.xcworkspace/xcshareddata/swiftpm), which corresponds to the lockfile for the Package Dependencies defined in the Xcode projects from `Orangina.xcworkspace` (see `XCSwiftPackageProductDependency` in `Orangina.xcodeproj/project.pbxproj`)
 - [`Components/Package.resolved`](https://github.tumblr.net/Tumblr/ios/blob/develop/Components/Package.resolved), which corresponds to the lockfile for [`Components/Package.swift`](https://github.tumblr.net/Tumblr/ios/blob/develop/Components/Package.swift)

Such a setup, with two `Package.resolved` in the repo, is totally valid, as long as each package definition source (either `Package.swift` file or `XCSwiftPackageProductDependency` entry in an Xcode project's `pxbproj`) has only one matching `Package.resolved` corresponding to that package.

Reply:
> If I'm understanding the way some repos use SPM, I think it might still be totally valid to have multiple Package.resolved files in the repo, when you have multiple (local) SwiftPM packages in a monorepo

Yep, that was also my understanding. Isn't the list of separate roots (e.g. `Components` and `Orangina.xcworkspace`) covering that scenario? 🤔

Reply:
Not given your current implementation iiuc because you error if the total number of Package.resolved files in the provided paths is > 1 while in those situations it would be expected to be one per provided path, not one total.

So maybe it's just your implementation that is incorrect and should be testing for `== spm_root_paths.length` not for `== 1`